client/app/components/faq/faqlist.js

```diff
@@ -35,94 +30,97 @@ class FaqList extends Component {
 			.then(() => console.log("Finished mounting!"))
+			// eslint-disable-next-line no-console
 			.catch(err => console.error(err));
 	}
 
 	componentWillReceiveProps(nextProps) {
 		// Avoiding refresh if possible
-	  if (nextProps.currentQuestions !== this.state.currentQuestions) {
-		 this.setState({ currentQuestions: nextProps.currentQuestions });
-	  }
+		if (nextProps.currentQuestions !== this.state.currentQuestions) {
+			this.setState({ currentQuestions: nextProps.currentQuestions });
+		}
 	}
 
-	render() {
-		return (
-			<ScrollView
-				style={FaqStyles.faqPadding, {backgroundColor: 'white'}}
-				refreshControl={
-					<RefreshControl
-						refreshing={this.state.isRefreshing}
-						onRefresh={this._onRefresh.bind(this)}
-					/>
-				}
-			>
-				<FlatList
-					data={this.state.currentQuestions}
-					renderItem={this._renderListItem}
-					extraData={this.state}
-					keyExtractor={this._keyExtractor}
-					ListHeaderComponent={this._renderHeader}
-				/>
-			</ScrollView>
-		);
+	onRefresh = () => {
+		this.setState({isRefreshing: true});
+		this.fetchData().then(() => {
+			this.setState({isRefreshing: false});
+		});
 	}
 
 	fetchData() {
 		return fetch(`${API_URL}/list`)
 			.then(response => response.json())
 			.then(faqList => this.props.onFAQLoaded(faqList))
+			// eslint-disable-next-line no-console
 			.catch(err => console.error(err));
 	}
 
-	_onRefresh() {
-		this.setState({isRefreshing: true});
-		this.fetchData().then(() => {
-			this.setState({isRefreshing: false});
-		});
-	}
-
-	_renderListItem = ({ item, index }) => {
+	renderListItem({ item, index }) {
 		let rowBg = index % 2 == 1 ? FaqStyles.faqListItemOdd : null;
+		
+		const icon = (
+			<Icon
+				name="chevron-thin-right"
+				size={30}
+				color="#787878"
+				style={{marginTop: 5}}
+			/>
+		);
+
 		return (
 			<ListItem
-				containerStyle={FaqStyles.faqListItem, rowBg}
+				containerStyle={{...FaqStyles.faqListItem, rowBg}}
 				titleStyle={FaqStyles.faqListItemText}
 				key={item.id}
 				title={item.question}
-				onPress={() => this._renderFaqDetails(item, index)}
-				rightIcon={<Icon
-								name={'chevron-thin-right'}
-								size={30}
-								color={'#787878'}
-								style={{marginTop: 5}}
-						  />}
+				onPress={this.renderFaqDetails(item, index)}
+				rightIcon={icon}
 			/>
 		);
 	}
 
-	_renderHeader = () => {
+	renderHeader = () => {
 		//TODO: Add actual search
 		return <SearchBar placeholder="Search for questions here!" lightTheme />;
 	}
 
-	_renderFaqDetails = (question, index) => {
-		this.props.navigation.navigate('FaqDetails', {
-			index: index,
-			currentQuestion: question,
-			questionList: this.state.currentQuestions,
-		});
+	renderFaqDetails = (question, index) => {
+		return () => {
+			this.props.navigation.navigate('FaqDetails', {
+				index: index,
+				currentQuestion: question,
+				questionList: this.state.currentQuestions,
+			});
+		}
 	}
-}
-
-FaqList.propTypes = {
```

Comment:
We should make a decision on using propTypes vs not using them.

Reply:
We should be using propTypes and defaultProps

Reply:
Let's do it! I'll temporarily remove them because I think they aren't very well laid out, but we can go through after all the components are sorted and add them. I think it would make a great beginner task too! What do you think?

Reply:
Let's leave them in so that whoever picks it up can refer to it.  There will be some code that's done for some components and not others because I wanted to create a template which other devs can copy to implement throughout.